tools/cgdiff/include/NodeSummary.h

```diff
@@ -0,0 +1,206 @@
+            oss << "[";
+            for (auto it = container.begin(); it != container.end(); ++it) {
+                oss << *it;
+                if (std::next(it) != container.end()) oss << ", ";
+            }
+            oss << "]";
+            return oss.str();
+        };
+
+        std::ostringstream os;
+
+        for (const auto& nd : nodeDiffs) {
+            os << nd.name << ":\n"
+                << "\ttype : " << printList(nd.diffType) << ", \n"
+                << "\tcalleesOnlyInA : " << printList(nd.calleesOnlyInA) << ", \n"
+                << "\tcalleesOnlyInB : " << printList(nd.calleesOnlyInB) << ", \n"
+                << "\tmetadataOnlyInA : " << printList(nd.metadataOnlyInA) << ", \n"
+                << "\tmetadataOnlyInB : " << printList(nd.metadataOnlyInB) << "\n\n";
+        }
+
+        return os.str();
+    }
+
+    /**
+     * Emit json representation of a list of NodeDiffs.
+     *
+     * @param nodeDiffs Vector of NodeDiff objects to format
+     * @param ignoring  List of difference categories that were ignored
+     * @return Formatted string summarizing the differences
+     */
+    static nlohmann::ordered_json emitAsJson(const std::vector<NodeDiff>& diffs, const std::vector<std::string>& ignoring) {
+        nlohmann::ordered_json j;
+        j["ignoring"] = ignoring;
+
+        for (const auto& diff : diffs) {
+            nlohmann::ordered_json inner;
+            inner["diffType"] = diff.diffType;
+            inner["calleesOnlyInA"] = diff.calleesOnlyInA;
+            inner["calleesOnlyInB"] = diff.calleesOnlyInB;
+            inner["metadataOnlyInA"] = diff.metadataOnlyInA;
+            inner["metadataOnlyInB"] = diff.metadataOnlyInB;
+
+            j[diff.name] = inner;
+        }
+
+        nlohmann::ordered_json root;
+        root["diff"] = j;
+        return root;
+    }
+};
+
+/**
+ * Create a NodeDiff representing the differences between two nodes.
+ *
+ * Comparison can be customized using the `mode` flags to ignore specific parts.
+ *
+ * @param nsA NodeSummary from the first graph
+ * @param nsB NodeSummary from the second graph
+ * @param mode ComparisonMode flags controlling which differences to ignore
+ * @return NodeDiff describing the differences between nsA and nsB
+ */
+NodeDiff createNodeDiff(const NodeSummary& nsA, const NodeSummary& nsB, ComparisonMode mode) {
+    NodeDiff diff;
+    diff.name = nsA.name;
+
+    if (!hasFlag(mode, ignoreBody) && nsA.hasBody != nsB.hasBody) {
+        diff.diffType.push_back("differentBody");
+    }
+
+    if (!hasFlag(mode, ignoreEdges) && nsA.callees != nsB.callees) {
+        diff.diffType.push_back("differentEdges");
+
+        for (const auto& callee : nsA.callees) {
+            if (!nsB.callees.count(callee)) {
+                diff.calleesOnlyInA.insert(callee);
+            }
+        }
+        for (const auto& callee : nsB.callees)
+            if (!nsA.callees.count(callee))
+                diff.calleesOnlyInB.insert(callee);
```

Comment:
```suggestion
            if (!nsA.callees.count(callee)) {
                diff.calleesOnlyInB.insert(callee);
            }
```
I think we always use curlys